server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/TabletBase.java

```diff
@@ -229,7 +235,13 @@ public Tablet.LookupResult lookup(List<Range> ranges, List<KVEntry> results,
   Batch nextBatch(SortedKeyValueIterator<Key,Value> iter, Range range, ScanParameters scanParams)
       throws IOException {
 
-    // log.info("In nextBatch..");
+    while (context.getLowMemoryDetector().isRunningLowOnMemory(context, DetectionScope.SCAN, () -> {
+      return isUserTable;
+    }, () -> {
+      log.info("Not starting next batch because low on memory, extent: {}", extent);
+      server.getScanMetrics().incrementScanPausedForLowMemory();
+      Uninterruptibles.sleepUninterruptibly(500, TimeUnit.MILLISECONDS);
```

Comment:
Would preventing interrupts delay shutdown?  Basically, would it be better to allow interrupts and stop or at least throw it up the stack?   This is a common question for all usages of `sleepUninterruptibly` that follow.

Reply:
Addressed in cacc718